crates/store/src/server/api.rs

```diff
@@ -265,6 +268,8 @@ impl api_server::Api for StoreApi {
         // TODO: Why the error is swallowed here? Fix or add a comment with explanation.
         let _ = self.state.apply_block(block).await;
 
+        self.block_storage.save_block(block_num, &request.block).await?;
```

Comment:
Should this not be a part of `state.apply_block()`? The main reason is that if `state.apply_block()` succeeds, but `block_storage.save_block()` fails, we'll end up in an inconsistent state.

The downside is that we'll need to serialize the block (instead of just getting `request.block`) but maybe that's OK?